### ISSUE TITLE
Generate zipped topic content

### ISSUE THREAD

**Issue by jmatsushita** [iilab/openmentoring-curation#1]
critical information to know per card: card type, stack title, ordinal within unit, content (html string), stack title?, ordinal within stack? … we should think about what we might want to display as we’re going from card to card, but this is probably a good start

**Comment by bsenyk** [comment]
The idea here is that, although we download content at the topic level, we should be able to only load the unit that somebody is interested in viewing.  With this goal in mind, here is how I would like to structure the zip file:

```
topic-slug.zip
  /unit-slug
    index.json   //maybe we call this unit.json to be less confusing?
    /assets
      unit-image-name-1.png
      unit-image-name-2.jpg
  /unit-slug-2
    index.json
    /assets
      unit-image-name-1.jpg
      unit-image-name-2.jpg
```

The index.json (or unit.json if we decide to name it that) for each unit could then look something like this:

``` javascript
{
  "slug": "practice-safe-social-networks_access-friends-contacts",
  "type": "unit",
  "title": "How to control access to friends &amp; contacts?",
  "parentTitle": "Safe Social Networks",
  "locale": "en",
  "categories": [],
  "updatedAt": "2015-11-26T00:00:00.000Z",
  "cards": [
    {
      "type": "<card-type-string>",
      "stackSlug": "1-intro",
      "stackTitle": "Intro",          //is this a thing?
      "content":"Welcome! <img src=\"assets/unit-image-name-2.jpg\" /> blah blah",
      "stackOrder": 1              //ordinal within this stack
    }
  ]
}
```

**Alternatively** we could follow a nested syntax for stacks and cards.  The advantage of this would be that ordinals within a stack would be evident based on array index, and within a unit would be easily calculable:

``` javascript
{
  "slug": "practice-safe-social-networks_access-friends-contacts",
  "type": "unit",
  "title": "How to control access to friends &amp; contacts?",
  "parentTitle": "Safe Social Networks",
  "locale": "en",
  "categories": [],
  "updatedAt": "2015-11-26T00:00:00.000Z",
  "stacks": [
    {
      "slug": "1-intro",
      "title": "Intro",          //is this a thing?
      "cards": [
         {
           "type": "<card-type-string>",
           "content":"Welcome! <img src=\"assets/unit-image-name-2.jpg\" /> blah blah"
         }
      ]
    }
  ]
}
```

**Comment by jmatsushita** [comment]
@bsenyk why do we need stack ordinals if stacks aren't visible for users? We also need categories in the cards don't we? My existing implementation is closer to your first proposal so I'll continue with it.

**Comment by bsenyk** [comment]
@jmatsushita On stack ordinals, I was thinking that even if we don't show visible stacks, there might be the concept of a section title, like "Intro" even if it spans multiple cards.  The ordinal probably still isn't necessary though -- a stackTitle would accomplish that

**Comment by jmatsushita** [comment]
@bsenyk Agreed on not needing ordinals for stacks (given we're by convention using file names that have ordinals in them for cards) but we need ordinals for units within topics, and topics within the index. Will work on this soon.

**Comment by bsenyk** [comment]
@jmatsushita Another option would be to un-flatten the structure a bit.  Have a units array within each topic in the index.  I end up structuring it like that for display anyways, so I'd be deleting code instead of adding it to support this.  It's totally up to you, though -- I can work with the existing structure + ordinals or a nested structure.
